Robust_Lab.qmd

```diff
@@ -128,10 +341,69 @@ df <- data.frame(hours=c(1, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4,
 
 a\. Use the lm() function to fit a regression model in R that uses **hours** as the predictor variable and **score** as the response variable
 
+```{r}
+mod = lm(score ~ hours, data = df)
+
+mod %>% 
+  broom::tidy() %>%
+  gt()
+```
+
 b\. Interpret the results
 
-c\. Check assumptions and report which ones failed (include plots)
+**The results of the model suggest that exam scores increase by 1.95 for each additional hour spent studying. However, the increase is not significant (p = 0.09).**
+
+**Marginal effect**
+
+c\. Check assumptions and report which assumptions are violated (include stats or plots)
+
+## (Residual) Linearity and Homoscedasticity
+
+```{r}
+#1 - Residual linearity and homoscedasticity
+performance::check_model(mod, check = c("linearity", "homogeneity"))
+```
+
+**The plots above suggest that the assumptions of (residual) linearity and homoscedasticity are not met by the data. The residuals and the predicted values do not fall along a straight line. In other words, the model has larger errors at different predicted values. The violation of the linearity assumption can also be read from a scatter plot of the *raw* values of the DV and IV:**
```

Comment:
if you run a test for these, they do not come back as being violated.  They just appear as such.

You can run performance::check_homoscedasticity() it will tell you if it is violated or not.

Reply:
Weird: when I run the following code, it returns a p value < .001. Does that not mean that the assumption is violated (i.e., that residuals are unequally distributed across the levels of the IV)? What am I missing?

`performance::check_heteroscedasticity(mod)`

I've added this to the script.